src/extrinsic/staking.rs

```diff
@@ -49,31 +60,49 @@ pub type StakingWithdrawUnbondedFn = (CallIndex, u32);
 pub type StakingNominateFn = (CallIndex, Vec<GenericAddress>);
 pub type StakingChillFn = CallIndex;
 pub type StakingSetControllerFn = (CallIndex, GenericAddress);
-
-pub type StakingBondXt = UncheckedExtrinsicV4<StakingBondFn>;
-pub type StakingBondExtraXt = UncheckedExtrinsicV4<StakingBondExtraFn>;
-pub type StakingUnbondXt = UncheckedExtrinsicV4<StakingUnbondFn>;
-pub type StakingRebondXt = UncheckedExtrinsicV4<StakingRebondFn>;
-pub type StakingWithdrawUnbondedXt = UncheckedExtrinsicV4<StakingWithdrawUnbondedFn>;
-pub type StakingNominateXt = UncheckedExtrinsicV4<StakingNominateFn>;
-pub type StakingChillXt = UncheckedExtrinsicV4<StakingChillFn>;
-pub type StakingSetControllerXt = UncheckedExtrinsicV4<StakingSetControllerFn>;
+pub type StakingPayoutStakersFn = (CallIndex, PayoutStakers);
+pub type StakingForceNewEraFn = (CallIndex, ForceEra);
+pub type StakingForceNewEraAlwaysFn = (CallIndex, ForceEra);
+pub type StakingForceNoEraFn = (CallIndex, ForceEra);
+pub type StakingSetPayeeFn = (CallIndex,GenericAddress);
+pub type StakingSetValidatorCountFn = (CallIndex, u32);
+
+pub type StakingBondXt<SignedExtra> = UncheckedExtrinsicV4<StakingBondFn, SignedExtra>;
+pub type StakingBondExtraXt<SignedExtra> = UncheckedExtrinsicV4<StakingBondExtraFn, SignedExtra>;
+pub type StakingUnbondXt<SignedExtra> = UncheckedExtrinsicV4<StakingUnbondFn, SignedExtra>;
+pub type StakingRebondXt<SignedExtra> = UncheckedExtrinsicV4<StakingRebondFn, SignedExtra>;
+pub type StakingWithdrawUnbondedXt<SignedExtra> =
+    UncheckedExtrinsicV4<StakingWithdrawUnbondedFn, SignedExtra>;
+pub type StakingNominateXt<SignedExtra> = UncheckedExtrinsicV4<StakingNominateFn, SignedExtra>;
+pub type StakingChillXt<SignedExtra> = UncheckedExtrinsicV4<StakingChillFn, SignedExtra>;
+pub type StakingSetControllerXt<SignedExtra> =
+    UncheckedExtrinsicV4<StakingSetControllerFn, SignedExtra>;
+pub type StakingPayoutStakersXt<SignedExtra> =
+    UncheckedExtrinsicV4<StakingPayoutStakersFn, SignedExtra>;
+pub type StakingForceNewEraXt<SignedExtra> =
+    UncheckedExtrinsicV4<StakingForceNewEraFn, SignedExtra>;
+pub type StakingForceNewEraAlwaysXt<SignedExtra> =
+    UncheckedExtrinsicV4<StakingForceNewEraAlwaysFn, SignedExtra>;
+pub type StakingForceNoEraXt<SignedExtra> = UncheckedExtrinsicV4<StakingForceNoEraFn, SignedExtra>;
+pub type StakingSetPayeeXt<SignedExtra> = UncheckedExtrinsicV4<StakingSetPayeeFn, SignedExtra>;
+pub type StakingSetValidatorCountXt<SignedExtra> =UncheckedExtrinsicV4<StakingSetValidatorCountFn, SignedExtra>;
 
 // https://polkadot.js.org/docs/substrate/extrinsics#staking
-impl<P, Client> Api<P, Client>
+impl<P, Client, Params> Api<P, Client, Params>
```

Comment:
To me it looks like you only need the `SignedExtra`, not the full `Params` trait. Is there a reason you're taking the  Params as input?

Reply:
I couldn't make it as only signedExtra because of it is a type inside the trait, any suggestion?

Reply:
Leave as is for now. It's fine. Thanks for looking into it 👍